js/guestshare.js

```
@@ -0,0 +1,142 @@
+ *
+ */
+
+var GuestShare = {
+	model: null,
+	email: null,
+
+	addGuest: function (model, email) {
+		this.model = model;
+		this.email = email;
+
+		var self = this;
+		var xhrObject = {
+			type: 'PUT',
+			url: OC.generateUrl('/apps/guests/users'),
+			dataType: 'text',
+			data: {
+				displayName: this.email,
+				username: this.email,
+				email: this.email
+			}
+		};
+
+		$.ajax(xhrObject).done(function (xhr) {
+			self._addGuestShare();
+		}).fail(function (xhr) {
+			var response = JSON.parse(xhr.responseText);
+			var error = response.errorMessages;
+			OC.dialogs.alert(
+				error.email, // text
+				t('core', 'Error') // title
+			);
+		});
+	},
+
+	_addGuestShare: function () {
+		var self = this;
+		var attributes = {
+			shareType: 0,
+			shareWith: this.email,
+			permissions: OC.PERMISSION_CREATE | OC.PERMISSION_UPDATE | OC.PERMISSION_READ | OC.PERMISSION_DELETE,
+			path: this.model.fileInfoModel.getFullPath()
+		};
+
+		return $.ajax({
+			type: 'POST',
+			url: OC.linkToOCS('apps/files_sharing/api/v1', 2) + 'shares?format=json',
+			data: attributes,
+			dataType: 'json'
+		}).done(function () {
+			if (self.model) {
+				self.model.fetch();
+			}
+		}).fail(function () {
+			OCdialogs.alert(
+				t('core', 'Error while sharing'), // text
+				t('core', 'Error') // title
+			);
+		});
+	},
+};
+
+OC.Plugins.register('OC.Share.ShareDialogView', {
+	attach: function (obj) {
+
+		// Override ShareDialogView
+		var oldHandler = obj.autocompleteHandler;
+		obj.autocompleteHandler = function(search, response) {
+
+		    return oldHandler.call(obj, search, function(result) {
+				var searchTerm = search.term.trim();
+
+				// Add potential guests to the suggestions
+				if (searchTerm.search("@") !== -1) {
+					result.push({
+						label: t('core', 'Add {unknown} (guest)', {unknown: searchTerm}),
+						value: {
+							shareType: 4,
+							shareWith: searchTerm
+						}
+					});
+				}
+
+				if (result !== undefined) {
```

please move this check at the begining of the function. Because if `result` is undefined, the `result.push` above will fail hard